aip/auth/4119.md

```diff
@@ -0,0 +1,135 @@
+federation policy in the Security Token Service (STS). The instructions on how
+to do this are out of scope of this AIP.
+
+### Using mTLS Token Binding
+
+The auth libraries **must** support the following values in the
+**"~/.config/gcloud/certificate_config.json"** configuration file. Note that the
+default location of this file can be changed using the
+`GOOGLE_API_CERTIFICATE_CONFIG` environment variable.
+
+```json
+{
+  “version”: 1
+  "cert_info": {
+    "workload": {
+      "cert_path": "path/to/cert/file"
+      "key_path": "path/to/key/file"
+      “workload_identity_provider”: “...”
+      “service_account_email”: “...”
+    },
+    "keychain": {
+      ...
+    },
+    "pkcs11": {
+      ...
+    },
+    "windows": {
+      ...
+    },
+  },
+  "libs": {
+   ...
+  }
+}
+```
+
+The following lists the fields relevant to mTLS token binding configuration:
+
+  - **"workload_identity_provider"**: The specified value will be used to
+    populate the request to STS to request identity-bound access tokens. This
+    value refers to the fully qualified name of the workload pool configured in
+    the federation policy in STS. The specified value **must** be of the
+    following format.
+
+```
+"workload_identity_provider":"//iam.googleapis.com/projects/<project_number>/locations/global/workloadIdentityPools/<pool_identifier>/providers/<provider_identifier>"
+```
+
+  - **"service_account_email"**: If set, the specified value will be used to
+    populate the request to IAM Credentials service to request identity-bound
+    access tokens. This value refers to the service account email to be used for
+    resource access. If not set, the service account email will be determined
+    automatically by querying the following Metadata Service endpoint:
+    `http://metadata/computeMetadata/v1/instance/service-accounts/default/email`.
+
+The description of the **"cert_path"** and **"key_path"** fields can be found in
+[Mutual Authentication Using Workload Credentials][2].
+
+To enable using token binding when communicating with Google APIs the following
+conditions are required:
+
+  - [Mutual Authentication Using Workload Credentials][2] **must** be enabled.
+  - The **"workload_identity_provider"** **must** be present and
+    **"service_account_email"** values **should** be present in the
+    **"workload"** section of the **"~/.config/gcloud/certificate_config.json"**
+    configuration file.
+
+### Expected Behavior
+
+To support the usage of identity-bound access tokens, the auth libraries
+**must** follow the steps below when sending requests to Google APIs:
+
+  1. Connect to the mTLS endpoint of the [Security Token Service (STS) API][3]
+     using the workload credentials provisioned as described in [Mutual
+     Authentication Using Workload Credentials][2]. This endpoint must be
+     `sts.mtls.googleapis.com`.
+  1. Send an HTTP request to STS’s ExchangeToken API requesting an
```

Comment:
nit: Should we say "Send a request" instead of "Send an HTTP request"?

Rationale: Using HTTP is an implementation detail rather than a requirement.

Similarly below for the bullet point about calling the IAM Credentials service.

Reply:
I think this AIPs are for developers as well. Having this detail may be useufl.

@shinfan WDYT?

Reply:
SG, thanks!